From a7a7df085edc0f4a7625dadc3d0c9862b4fa4bce Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:06 -0400
Subject: [powerpc] pseries: Fix SMP=n build of rng.c

Message-id: <1410545655-205645-198-git-send-email-dzickus@redhat.com>
Patchwork-id: 94629
O-Subject: [RHEL7 PATCH 197/626] powerpc/pseries: Fix SMP=n build of rng.c
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 148924f7a282b3acb1f8868ae03c3f76be301d8d
Author: Michael Ellerman <mpe@ellerman.id.au>
Date:   Wed Nov 20 11:05:02 2013 +1100

    powerpc/pseries: Fix SMP=n build of rng.c

    In commit a489043 "Implement arch_get_random_long() based on H_RANDOM" I
    broke the SMP=n build. We were getting plpar_wrappers.h via spinlock.h
    which breaks when SMP=n.

    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/rng.c b/arch/powerpc/platforms/pseries/rng.c
index a702f1c..72a1027 100644
--- a/arch/powerpc/platforms/pseries/rng.c
+++ b/arch/powerpc/platforms/pseries/rng.c
@@ -13,6 +13,7 @@
 #include <linux/of.h>
 #include <asm/archrandom.h>
 #include <asm/machdep.h>
+#include <asm/plpar_wrappers.h>
 
 
 static int pseries_get_random_long(unsigned long *v)
-- 
1.7.1